config.py

```diff
@@ -0,0 +1,102 @@
+from configparser import ConfigParser
+
+class GlobalOptions():
+    ''' Container class for global options
+    '''
+    def __init__(self, port, mcast_link, mcast_site, default_domain, default_domain_type, ipv4_gateway):
+        self.port = port
+        self.mcast_link = mcast_link
+        self.mcast_site = mcast_site
+        self.default_domain = default_domain
+        self.default_domain_type = default_domain_type
+        self.ipv4_gateway = ipv4_gateway
+
+class DomainOptions():
+    ''' Base container class for per domain options
+    '''
+    @classmethod
+    def from_parser(cls, section, parser, globals):
+        ''' Builds a DomainOptions object from a config section
+            Handles domain type specific options automatically
+        '''
+        from domain import DomainType
+
+        type = parser.get(section, 'DomainType', fallback=globals.default_domain_type)
+        # Get DomainOptions subclass for type and instantiate
+        return DomainType.get(type.lower()).options(section, parser, globals)
```

Comment:
Using the fallback value does not seem to work.

```py3tb
Traceback (most recent call last):
  File "/opt/mesh-announce/respondd.py", line 71, in <module>
    config = Config.from_file(args.config)
  File "/opt/mesh-announce/config.py", line 61, in from_file
    return cls(parser)
  File "/opt/mesh-announce/config.py", line 69, in __init__
    self._initialize_domain_options(parser, domain)
  File "/opt/mesh-announce/config.py", line 88, in _initialize_domain_options
    self.domains[domain] = DomainOptions.from_parser(domain, parser, self.globals)
  File "/opt/mesh-announce/config.py", line 26, in from_parser
    return DomainType.get(type.lower()).options(section, parser, globals)
AttributeError: 'NoneType' object has no attribute 'lower'
```

Reply:
Yeah, I see. The domain type fallback should work properly now